src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -1738,9 +1742,65 @@ AS
 
         -- Add the new tags with the given status
         INSERT INTO dbo.ExtendedQueryTag
-            (TagKey, TagPath, TagPrivateCreator, TagVR, TagLevel, TagStatus)
+            (TagKey, TagPath, TagPrivateCreator, TagVR, TagLevel, TagStatus, QueryStatus)
         OUTPUT INSERTED.TagKey
-        SELECT NEXT VALUE FOR TagKeySequence, TagPath, TagPrivateCreator, TagVR, TagLevel, @ready FROM @extendedQueryTags
+        SELECT NEXT VALUE FOR TagKeySequence, TagPath, TagPrivateCreator, TagVR, TagLevel, @ready, 1 FROM @extendedQueryTags
+
+    COMMIT TRANSACTION
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     UpdateExtendedQueryTagQueryStatus
+--
+-- DESCRIPTION
+--    Update QueryStatus of extended query tag
+--
+-- PARAMETERS
+--     @tagPath
+--         * The extended query tag path
+--     @queryStatus
+--         * The query  status
+--
+-- RETURN VALUE
+--     The modified extended query tag.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.UpdateExtendedQueryTagQueryStatus (
+    @tagPath VARCHAR(64),
+    @queryStatus TINYINT
+)
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    BEGIN TRANSACTION
+
+        DECLARE @tagKey INT
+        DECLARE @currentQueryStatus TINYINT
+        DECLARE @currentTagStatus TINYINT
+
+        -- Check if tag exists
+        SELECT @tagKey = TagKey, @currentQueryStatus = QueryStatus , @currentTagStatus = TagStatus
+        FROM dbo.ExtendedQueryTag WITH (HOLDLOCK)
+        WHERE TagPath = @tagPath
+
+        IF @@ROWCOUNT = 0
+            THROW 50404, 'extended query tag is not found', 1
+
+        -- Check if tag is in Ready status
+        -- 0 - Adding, 1 - Ready, 2 - Deleting
+        IF @currentTagStatus <> 1
+            THROW 50409, 'extended query tag is not in Ready status', 1
+
+        -- Check if tag queryStatus is same as input
+        IF @currentQueryStatus = @queryStatus
+            THROW 50409, 'extended query tag is in required status already', 2
```

Comment:
Do we really care if this is the case? If they change it to be the same value, then no problem! This doesn't adversely affect us. It also means we don't need to add all of the exception code 😛